k9mail/src/main/java/com/fsck/k9/service/NotificationActionService.java

```diff
@@ -34,37 +35,83 @@
     private final static String EXTRA_ACCOUNT = "account";
     private final static String EXTRA_MESSAGE = "message";
     private final static String EXTRA_MESSAGE_LIST = "messages";
+    private final static String EXTRA_DONTCANCEL = "dontcancel";
+    /**
+     * ID of the notification that triggered an intent.
+     * Used to cancel exactly that one notification because due to
+     * Android Wear there may be multiple notifications per account.
```

Comment:
If that's the cade, you'll need to adapt the requestCodes of the PendingIntent creations, as those are what keep the intent extras in place; otherwise you'll still end up with one notification per account.